climada/hazard/test/test_base.py

```diff
@@ -212,6 +212,47 @@ def test_get_date_strings_pass(self):
         self.assertEqual(haz.get_event_date()[560],
                          u_dt.date_to_str(haz.date[560]))
 
+    def test_matrix_update(self):
+        """Check that the csr_matrix can be updated with element access"""
+        self.hazard.intensity[[0, 2], 1] = 10
+        np.testing.assert_array_equal(
+            self.hazard.intensity.toarray(), [[1, 10], [1, 2], [1, 10]]
+        )
+        self.hazard.fraction[:, 1] = 0
+        np.testing.assert_array_equal(
+            self.hazard.fraction.toarray(), [[1, 0], [1, 0], [1, 0]]
+        )
+
+    def test_matrix_consistency(self):
+        """Check that the csr_matrix is brought in canonical format"""
+        # Non-canonical: First three data points will be summed onto the first matrix
+        # entry, forth will be an explicit zero entry
+        data = [0, 1, 2, 0]
+        indices = [0, 0, 0, 1]
+        indptr = [0, 4, 4, 4]
+        matrix = sparse.csr_matrix((data, indices, indptr), shape=(3, 2))
+        np.testing.assert_array_equal(matrix.data, data)
+        np.testing.assert_array_equal(matrix[0, [0, 1]].toarray(), [[3, 0]])
+        self.assertEqual(matrix.nnz, 4)
+        self.assertFalse(matrix.has_canonical_format)
```

Comment:
I am not sure we want to test the `has_canonical_format method` from scipy. I would just remove this line.

Reply:
I left it in the test for `prune_matrix`, just to be sure that we are using the `csr_matrix` as intended, and things are indeed working as expected.